From a88d5079038417d21f2f492e4324e697f2ec52b9 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Wed, 3 Jun 2015 19:38:28 -0400
Subject: [PATCH] Expose CommandMap

Many plugins prefer to implement their own flexible command system,
and avoid using the plugin.yml for command registration. Currently
this requires Reflection to access the command map to dynamically
register commands.

This commit exposes the Command Map so that you may register a
command without defining it in plugin.yml. This is extremely useful
to plugins that have a lot of commands and also want to dynamically
register commands based on context (such as plugin being used by
multiple servers, but certain commands only for certain server types).
---
 src/main/java/org/bukkit/Server.java | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index b991a14..1886a65 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -14,6 +14,7 @@ import java.util.logging.Logger;
 
 import org.bukkit.Warning.WarningState;
 import org.bukkit.command.CommandException;
+import org.bukkit.command.CommandMap;
 import org.bukkit.command.CommandSender;
 import org.bukkit.command.ConsoleCommandSender;
 import org.bukkit.command.PluginCommand;
@@ -359,11 +360,18 @@ public interface Server extends PluginMessageRecipient {
     /**
      * Gets a services manager.
      *
-     * @return s services manager
+     * @return a services manager
      */
     public ServicesManager getServicesManager();
 
     /**
+     * Returns the servers command map for managing commands.
+     *
+     * @return The servers command map
+     */
+    public CommandMap getCommandMap();
+
+    /**
      * Gets a list of all worlds on this server.
      *
      * @return a list of worlds
-- 
1.9.1

